### PR TITLE
feat: Add contributing document

### PR DESCRIPTION
**Pull request by TerryHowe**
This is a start at least.

Partial: https://github.com/oras-project/oras/issues/780

### CODE REVIEW

**Comment by FeynmanZhou** [comment]
@TerryHowe Agree that we can use it as a start. 
FYI there is a [CONTRIBUTING.md template](https://github.com/cncf/project-template/blob/main/CONTRIBUTING.md) provided by CNCF.

**Review by FeynmanZhou — APPROVED** [review]
LGTM. It can be used as a base version. We can iterate the content gradually.

**Review by SteveLasker — APPROVED** [review]
LGTM
Thanks @TerryHowe for taking the time to add this info to help folks get started

**Review by vsoch — COMMENTED** [review]
This looks good - my only suggestions:

- include a full example for committing and signing
- all links might formatted as links (e.g., developercertificate.org)
- Details about opening a PR against a branch? Process for the PR? Anything else relevant to that process? Do maintainers require rebase?

**Comment by sajayantony** [comment]
Agree with @FeynmanZhou here. Given that ORAS is under CNCF guidelines it makes sense to adopt the CNCF template.

**Review by Mr-DG-Wick — APPROVED** [review]
LGTM!

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  * [Contact Us](#contact-us)
+  * [Pull Request Lifecycle](#pull-request-lifecycle)
+  * [Development Environment Setup](#development-environment-setup)
+  * [Sign Your Commits](#sign-your-commits)
+  * [Pull Request Checklist](#pull-request-checklist)
+
+Welcome! We are glad that you want to contribute to the ORAS project! 💖
+
+As you get started, you are in the best position to give us feedback on areas of
+our project that we need help with including:
```
Comment (Mr-DG-Wick) [inline docs/contributing.md:16]:
```suggestion
our project that we need help with, including:
```

Comment (Mr-DG-Wick) [reply docs/contributing.md:16]: a minor grammatical fix.

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+Please come to our [meetings](#come-to-meetings) or [contact us](#contact-us) and let's discuss how we can work
+together.
+
+### Come to Meetings
+
+Absolutely everyone is welcome to come to any of our meetings. You never need an
+invitation to join us. In fact, we want you to join us, even if you don’t have
+anything you feel like you want to contribute. Just being there is enough!
+
+Join the weekly ORAS community meeting at 4:00 PM PST on [Zoom](https://zoom.us/j/8079328631?pwd=UzN0NUlmbmcvTTN5L0o5VUQ2YndyQT09) with ID 807 932 8631.
```
Comment (FeynmanZhou) [inline docs/contributing.md:47]: Note: weekly => bi-weekly. The meeting cadence changed from once a week to bi-weekly in 2023.

**Review by FeynmanZhou — COMMENTED** [review]
@TerryHowe  Just a small suggestion. Overall looks much better! Thank you for bringing a good contributing guide for ORAS.

File: docs/contributing.md (hunk 2 above)
Comment (TerryHowe) [reply docs/contributing.md:47]:
```suggestion
Join the bi-weekly ORAS community meeting at 4:00 PM PST on [Zoom](https://zoom.us/j/8079328631?pwd=UzN0NUlmbmcvTTN5L0o5VUQ2YndyQT09) with ID 807 932 8631.
```

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Comment by FeynmanZhou** [comment]
> This looks good - my only suggestions:
> 
> * include a full example for committing and signing
> * all links might formatted as links (e.g., developercertificate.org)
> * Details about opening a PR against a branch? Process for the PR? Anything else relevant to that process? Do maintainers require rebase?

It's nice to consider and bring these suggested content into the contribution guide in the second phase. I would suggest merging this PR first as a base version.

**Pull request merged**